libnavigation-base/src/main/java/com/mapbox/navigation/base/options/LocationPollingOptions.kt

```diff
@@ -0,0 +1,108 @@
+package com.mapbox.navigation.base.options
+
+import com.mapbox.android.core.location.LocationEngineRequest
+
+/**
+ * Defines options for the navigator polling loops.
+ *
+ * TODO move NavigationOptions.navigatorPredictionMillis to these options
+ * TODO move NavigationOptions.locationEngine to these options
```

Comment:
Consider the options here.

- [ ] Keep as is, with the task to make a SEMVER change in 2.0.0
- [ ] Move now, deprecate NavigationOptions. 🤔  but this would break the old options causing a SEMVER too
- [ ] Leave it as is
- [ ] Something else

Reply:
We've committed to exposing as few mutable options as possible which makes the SDK more stable and easier to maintain, that's why the majority of the customizable elements are provided via immutable `NavigationOptions`. I think we shouldn't break out of that pattern in v1. My preference would move these newly exposed fields to `NavigationOptions` and we can revisit this setup for v2.

If developers would like dynamic control over raw location requests (in the context of saving the battery that you mentioned below), they can always override the `LocaitonEngine`.

Reply:
https://github.com/mapbox/mapbox-navigation-android/pull/3800#discussion_r526497322